src/computation/logic/distribute-levelling-seats.ts

```diff
@@ -2,15 +2,94 @@ import { ComputationPayload, PartyResult, DistrictResult, PartyRestQuotients, Re
 import { Dictionary, dictionaryToArray } from "../../utilities/dictionary";
 import { distributeSeats } from ".";
 import { distributeLevelingSeatsOnDistricts, distributeLevelingSeatsOnDistrictsPre2005 } from "./utils";
+import { DistributionResult, NationalDistributionResult } from "../../computation/computation-models";
+import { isLargestFractionAlgorithm, isQuotientAlgorithm } from "./algorithm-utilities";
 
+/**
+ * Distributes the leveling seats on the parties and on each district.
+ *
+ * @param payload The computation payload
+ * @param partyResults The party results for each party
+ * @param districtPartyResults The party results for each district
+ * @param districtResults The district results
+ */
 export function distributeLevelingSeats(
     payload: ComputationPayload,
     partyResults: Dictionary<PartyResult>,
     districtPartyResults: Dictionary<Dictionary<PartyResult>>,
     districtResults: Dictionary<DistrictResult>
 ): PartyRestQuotients[] {
-    let levelingParties: Result[] = Object.keys(partyResults).map((partyCode) => {
-        return {
+    const allPartyCodes = [...Object.keys(partyResults)];
+    // Filter out parties with less than the threshold
+    const thresholdPartyCodes = allPartyCodes.filter(
+        (partyCode) => partyResults[partyCode].percentVotes >= payload.electionThreshold
+    );
+
+    const resultNationalFilter = nationalDistributionFilter(thresholdPartyCodes, payload, partyResults);
+
+    const levelingPartyCodes = resultNationalFilter.levelingPartyCodes;
+
+    const levelingSeatDistribution = finalLevelingSeatDistribution(
+        levelingPartyCodes,
+        partyResults,
+        payload,
+        resultNationalFilter.nationalDistribution
+    );
+
+    for (const partyCode in levelingSeatDistribution.seatsWon) {
+        if (levelingSeatDistribution.seatsWon.hasOwnProperty(partyCode)) {
+            const levelingSeats = levelingSeatDistribution.seatsWon[partyCode];
+            partyResults[partyCode].levelingSeats += levelingSeats;
+            partyResults[partyCode].totalSeats += levelingSeats;
+        }
+    }
+
+    const wonLevelingPartyCodes = levelingPartyCodes.filter((partyCode) => partyResults[partyCode].levelingSeats > 0);
+
+    let partyRestQuotients: Dictionary<PartyRestQuotients> = {};
+
+    if (payload.election.year < 2005) {
```

Comment:
Magic number still. This should probably be a const that we name and export given how often it's used and in how many places.

Reply:
I agree, earlier I asked you for a suggestion for a name. It is quite hard to find something that is intuitive and clear, rather than just more confusing.

Reply:
Well, if we stay in the scope of this bit of code without thinking about exports, couldn't we do `const preOneLevellingSeatPerDistrict = payload.election.year < 2005`? It's a little bulky but does make the intention quite clear.

Reply:
Sure, I was thinking more on a repository-wide parameter, since it is a conditional that appears in a lot of different places. But finding a name per scope is not quite as challenging.